tools/devtool

```diff
@@ -1432,6 +1448,70 @@ cmd_install() {
     install -m 755 "$CARGO_SECCOMPILER_TARGET_DIR/$target/$profile/seccompiler-bin" "$install_path"
 }
 
+cmd_build_kernel() {
+    nprocs=2
+
+    # Parse any command line args.
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            "-h"|"--help")      { cmd_help; exit 1; } ;;
+            "-c"|"--config")
+                shift
+                local KERNEL_CFG="$1"
+                ;;
+            "-n"|"--nproc")
+                shift
+                nprocs="$1"
+                ;;
+            *)
+                die "Unknown argument: $1. Please use --help for help."
+            ;;
+        esac
+        shift
+    done
+
+    # It is mandatory to provide a valid config file for building the kernel.
+    [ -z "$KERNEL_CFG" ] && die "You need to provide the path to a valid kernel config!"
+
+    arch=$(uname -m)
+    if [ "$arch" = "x86_64" ]; then
+        target="vmlinux"
+        cfg_pat="x86"
+        format="elf"
+    elif [ "$arch" = "aarch64" ]; then
+        target="Image"
+        cfg_pat="arm64"
+        format="pe"
+    else
+        die "Unsupported architecture!"
+    fi
+
+    kernel_dir_host="$FC_BUILD_DIR"/kernel
+    kernel_dir_ctr="$CTR_FC_BUILD_DIR"/kernel
+    create_dir "$kernel_dir_host"
+
+    # Extract the kernel version from the config file provided as parameter.
+    KERNEL_VERSION=$(cat "$KERNEL_CFG" | grep -Po "^# Linux\/$cfg_pat (([0-9]+.){2}[0-9]+)" | cut -d ' ' -f 3)
+    validate_kernel_version "$KERNEL_VERSION"
+
+    recipe_url="https://raw.githubusercontent.com/dianpopa/vmm-reference/fc_down/resources/kernel/make_kernel.sh"
```

Comment:
Indeed maybe I should have placed a comment here. The script that contains all the auxiliary methods for building a kernel is currently taken from my rust-vmm/vmm-reference fork since I did some changes there to work for aarch64. After [this PR is merged](https://github.com/rust-vmm/vmm-reference/pull/149), we will be able to download it directly from rust-vmm. The main goal here is to use as much as possible rust-vmm functionality.

Reply:
I understand. Still, how do we plan to consume it from rust-vmm ? Will they publish it to an official repository ? If we could install it as a python package in our docker image it would also be awesome. 

I have concerns with just copying the script from the github project even if the project is trusted. Especially in this case where we only probably need 10-20 lines of code, I would rather copy the code to our project instead of downloading a script.

Reply:
What do you mean by official repository? It is currently in vmm-reference. Do you mean like a repository that is only meant for building test resources? 

Making it available in our docker image is a really good idea indeed.

Reply:
Sorry, I mean a package registry. Something like `crates.io` for rust. I guess there must be something similar for python.